scripts/inactive_issue_checker.py

```
@@ -0,0 +1,232 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Inactive issue checker script for Oppia."""
+
+from __future__ import annotations
+
+import datetime
+import logging
+import os
+
+import requests
+from typing import List
+
+INACTIVE_DAYS = 7
+REPO_OWNER = 'oppia'
+REPO_NAME = 'oppia'
+
+
+class InactiveIssue:
+    """Represents an inactive issue with its number and assignee."""
+
+    def __init__(self, number: int, assignee: str):
+        self.number = number
+        self.assignee = assignee
+
+
+def get_inactive_issues(
+    github_token: str,
+    repo_owner: str,
+    repo_name: str
+) -> List[InactiveIssue]:
+    """Identifies inactive issues that need unassignment.
+
+    Args:
+        github_token: str. The GitHub token for authentication.
+        repo_owner: str. The owner of the repository.
+        repo_name: str. The name of the repository.
+
+    Returns:
+        List[InactiveIssue]. A list of issues that need unassignment.
+    """
+    headers = {
+        'Authorization': f'token {github_token}',
+        'Accept': 'application/vnd.github.v3+json',
+    }
+    repo_url = f'https://api.github.com/repos/{repo_owner}/{repo_name}'
+    inactive_issues: List[InactiveIssue] = []
+
+    issues_url = f'{repo_url}/issues?state=open'
+    response = requests.get(issues_url, headers=headers, timeout=10)
+    issues = response.json()
+
+    collaborators_url = f'{repo_url}/collaborators'
+    collaborators_response = requests.get(
+        collaborators_url, headers=headers, timeout=10
+    )
+    collaborators = collaborators_response.json()
+    collaborator_logins = set()
+    for c in collaborators:
+        collaborator_logins.add(c['login'])
+
+    pulls_url = f'{repo_url}/pulls?state=open'
+    pulls_response = requests.get(pulls_url, headers=headers, timeout=10)
+    pull_requests = pulls_response.json()
+
+    for issue in issues:
+        if not issue or not isinstance(
```

In what cases would `issue` not be a dict?

---

Although github API's JSON res gets converted into dict by python, but this is just for safer approach to avoid any failures.

---

If this response from an external API is something that we're not expecting, then it would be better to fail noisily so that we can report the error to GitHub if needed.

In general we should only handle errors that are expected to occur in practice.

---

@seanlip yeah there may be cases possbile when this reqeusts could fail, like due to rate limiting, internal server error, etc. So now I thought to handle this codeblock in a safe manner; so I propose to enclose them in a try catch block and log errors (if occur any) to caught any surprise activity. Adding on this `isinstance` will make sure type safety of incoming data. From my POV request could never fail but there is 10% chance of failure for which `try-catch` block is ready to handle it.